File: 118. Pascal's Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 1:
            return [[1]]
        if numRows == 2:
            return [[1], [1, 1]]
        output = [[1], [1, 1]]
        for i in range(2, numRows):
            tmp, fill = output[-1], [1, 1]
            for j in range(1, i):
                fill.insert(j, tmp[j - 1] + tmp[j])
            output.append(fill)
        return output

        # '''
        # 1
        # 1   1
        # 1   2   1
        # 1   3   3   1
        # 1   4   6   4   1
        # 1   5   10  10  5   1
        # .
        # .
        # .'''

File: 118. Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        for (int i=0;i<numRows;i++){
            vector<int> tmp(i+1, 1);
            if (i<=1){
                res.push_back(tmp);
            }else{
                for (int j=1;j<i;j++){
                    tmp[j] = res[i-1][j] + res[i-1][j-1];
                }
                res.push_back(tmp);
            }
        }
        return res;
    }
};
